omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java

```diff
@@ -81,12 +140,149 @@ public void setDefaultLocale(String defaultLocale) {
+	            String tmpProficientLocales = getProficientLocales();
+	            this.proficientLocalesSet = new HashSet<LocaleOption>();
+                for (Locale loc : displayLocales)
+                {
+                	LocaleOption locOption = new LocaleOption();
+                	locOption.locale = loc;
+                	locOption.language = loc.getDisplayName();
+                	locOption.name = loc.toString();
+                	this.proficientLocalesSet.add(locOption);
+                }
+	            setProficientLocales(tmpProficientLocales);
+            }
+        }
+
+        public Set<Locale> getDisplayLocales() {            
+            return displayLocales;
+        }
+
+        public void setDisplayLocales(Set<Locale> displayLocales) {
+            this.displayLocales = displayLocales;
+            if (displayLocales != null)
+            {
+	            String tmpProficientLocales = getProficientLocales();
+	            proficientLocalesSet = new HashSet<LocaleOption>();
+                for (Locale loc : displayLocales)
+                {
+                	LocaleOption locOption = new LocaleOption();
+                	locOption.locale = loc;
+                	locOption.language = loc.getDisplayName();
+                	locOption.name = loc.toString();
+                	proficientLocalesSet.add(locOption);
+                }
+	            setProficientLocales(tmpProficientLocales);
+        	}	
+        }
+        
+        private String[] getProficientLocalesList()
+        {
+        	if (proficientLocalesSet == null)
+        		return null;
+        	String[] tmpProfLocList = new String[proficientLocalesSet.size()];
+        	int idx = 0;
+            for (LocaleOption locOption : this.proficientLocalesSet)
+            {
+            	tmpProfLocList[idx] = locOption.getName();
+            }
+        	return tmpProfLocList;
+        }
+        
+        private void setProficientLocalesList(String[] proficientLocalesList)
+        {
+        	if (proficientLocalesList == null)
+        		return;
+        	clearProficientLocalesSet();
+            for (String loc : proficientLocalesList)
+            {
+            	if (loc == "false")
+            		continue;
+                for (LocaleOption locOption : this.proficientLocalesSet)
+                {
+                	if (locOption.getName().compareTo(loc) == 0)
+                	{
+                   		locOption.selected = true;
+                	}
+                 }
+            }
+        }
+        
+        private String getProficientLocalesFromSet() {
+        	if (proficientLocalesSet == null) 
+        		return "";
+        	StringBuilder retProficientLocales = new StringBuilder();
+            for (LocaleOption locOption : proficientLocalesSet)
+            {
+            	if (locOption.selected)
+            	{
+                	if (retProficientLocales.length() > 0)
+                		retProficientLocales = retProficientLocales.append(",");
```

Comment:
The cleaner way to handle this is to add a StringToLocaleConverter to uiframework module and there examples of these in the module, field binding by type using the @BindParam annotation would create for you this list and set on the UserDefault Object magically with minimal code written